lib/firestore_guild_setting.py

```diff
@@ -0,0 +1,52 @@
+class SettingSnapshot:
+    def __init__(self, document, setting):
+        self.document = document
+        self.setting = setting
+        self.bot = setting.bot
+
+    async def data(self):
+        result = await self.bot.loop.run_in_executor(self.executor, self.document.get)
+
+        return result.to_dict() or await self.create()
+
+    async def exists(self):
+        result = await self.bot.loop.run_in_executor(self.executor, self.document.get)
+
+        return result.exists
+
+    async def create(self):
+        if await self.exists():
+            return
+        payload = dict(
+            name=True,
+            emoji=True,
+            bot=False,
+            limit=100,
+            keep=True
+        )
+        await self.bot.loop.run_in_executor(self.executor, self.document.set, payload)
+        return payload
+
+    async def edit(self, name=None, emoji=None, bot=None, limit=None, keep=None):
+        data = await self.data()
+        name = name if name is not None else data['name']
+        emoji = emoji if emoji is not None else data['emoji']
+        bot = bot if bot is not None else data['bot']
+        limit = limit if limit is not None else data['limit']
+        keep = keep if keep is not None else data['keep']
```

Comment:
`None` は Falsy なので `None or 'abc'` の実行結果は `'abc'` です。
これを使って、以下のように改善できます。
```suggestion
        data = await self.data()
        name = name or data['name']
        emoji = emoji or data['emoji']
        bot = bot or data['bot']
        limit = limit or data['limit']
        keep = keep or data['keep']
```

Reply:
あたいがFalseになる可能性があって、その時元がTrueだと、False or TrueでTrueが優先されてしまうので、is not Noneにしてあります><
実際それで現在バグが発生しています...

Reply:
なるほどね。書きながら少し予想出来てた。